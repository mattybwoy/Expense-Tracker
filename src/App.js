import React, { useState } from 'react';
import Tracker from './Tracker'

function App() {
  return (
    < Tracker />
  )
}

export default App;