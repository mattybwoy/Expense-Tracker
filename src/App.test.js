import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event'


test('renders correctly when there are no expenses', () => {
  const tree = renderer.create(<App /> ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Allows expense to be deleted', () => {
  render(<App />)

    const textBox = screen.getByRole('textbox');
    userEvent.type(textBox, 'Book')
    //console.log(textBox)
    userEvent.click(screen.getByRole('spinbutton'), 5)
    //console.log(numberBox)

    userEvent.click(screen.getByText('Add'))
    const note = screen.getByDisplayValue('Book');
    userEvent.click(screen.getByRole('Delete'))

    expect(note).not.toBeInTheDocument
})
