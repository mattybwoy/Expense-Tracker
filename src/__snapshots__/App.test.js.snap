// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no expenses 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron"
    style={
      Object {
        "backgroundColor": "#bdbdbd ",
      }
    }
  >
    <h2
      className="text-center"
    >
      Expenses Tracker
      <br />
    </h2>
    <div
      className="text-center"
    >
      <p>
        Total Expenditure: 
        
        <span
          className="text success"
        >
          £
          
          0
        </span>
      </p>
    </div>
    <form
      className=""
      style={
        Object {
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      <div
        className="row form-group"
      >
        <label
          className="col-sm-4 col-form-label"
          htmlFor="exampleEmail"
        >
          Name of Expense
        </label>
        <div
          className="col-sm-4"
        >
          <input
            className="form-control"
            id="expenseName"
            name="name"
            onChange={[Function]}
            placeholder="Expense"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="row form-group"
      >
        <label
          className="col-sm-4 col-form-label"
          htmlFor="exampleEmail"
        >
          Amount £
        </label>
        <div
          className="col-sm-4"
        >
          <input
            className="form-control"
            id="expenseAmount"
            name="amount"
            onChange={[Function]}
            placeholder="0.00"
            role="spinbutton"
            type="number"
            value=""
          />
        </div>
      </div>
      <button
        aria-label={null}
        className="add btn btn-primary"
        onClick={[Function]}
        type="submit"
      >
        Add
      </button>
       
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        type="submit"
      >
        Reset
      </button>
    </form>
    <div>
      <ul
        className="list-group"
      />
    </div>
  </div>
  <footer>
    <strong>
      A MATTYBWOY CREATION
    </strong>
    <br />
    Powered by React
    <br />
    <img
      alt="https://reactjs.org/"
      src="logo.svg"
      width="40"
    />
  </footer>
</div>
`;
